e2e/errors/pump_errors.spec.js

```diff
@@ -0,0 +1,115 @@
+const { Test } = require('../../src/Test');
+const { Config } = require('../../src/config/index');
+
+describe('errors', () => {
+  var test = new Test();
+  var config = new Config();
+  var cgmData;
+  beforeAll(async () => {
+    config = await config.prepare();
+    let cgmText = config.text.device.CGMSimulatorScreen;
+    cgmData = {
+      model: { name: cgmText.Model.Constant, bgValues: [97] },
+      frequency: { seconds: true },
+      history: { name: cgmText.History.BackfillGlucose, backfillHours: 5 },
+    };
+  });
+  it('prepare test', async () => {
+    test = test.setup({
+      language: config.text,
+      screenDefaults: config.screenDefaults,
+      limits: config.limits,
+      authentication: { faceid: true },
+      enableClosedLoop: true,
+      warmupPeriod: { milliseconds: 10000 },
+      enableTherapySettings: true,
+      simulators: { cgm: true, pump: true },
+      cgmData,
+    });
+    await test.prepare();
+  });
+  describe('generate error on suspend', () => {
+    let pumpScreen;
+    let homeScreen;
+    beforeAll(async () => {
+        pumpScreen = await test.OpenPumpScreen();
+        await pumpScreen.Apply({ errorOnSuspend: true });
+        await pumpScreen.SuspendDeliveryButton.tap();
+    });
+    it('and check error shown on pump setting screen', async () => {
+        await pumpScreen.HasAlert();
+    });
+    it('then dismiss the error', async () => {
+        await pumpScreen.OKDismissAlertButton.tap();
+    });
+    it('and reset error on suspend', async () => {
+        await pumpScreen.Apply({ errorOnSuspend: false });
+        await pumpScreen.DoneButton.tap();
+    });
+    it('and check no error on status screen', async () => {
+        homeScreen = await test.OpenHomeScreen();
+        await expect(homeScreen.HeaderSection.PumpErrorLabel).toBeNotVisible();
+    });
```

Comment:
should this also check that the `Pump suspended` and `tap to resume` status is displayed?

Reply:
yes it absolutely should. I was thinking about going in and expanding these a bit. But I didn't want to do too much at once. If you're cool with it then I will definitely add a couple more check to make this a more complete test